nixos/modules/services/web-apps/healthchecks.nix

```
@@ -0,0 +1,216 @@
+{ config, lib, pkgs, buildEnv, ... }:
+
+with lib;
+
+let
+  cfg = config.services.healthchecks;
+  staticDir = cfg.dataDir + "/static";
+  boolToPython = b: if b then "True" else "False";
+  localSettingsFile = pkgs.writeTextFile {
+    name = "local_settings.py";
+    text = ''
+      STATIC_ROOT = '${staticDir}'
+      ALLOWED_HOSTS = ${builtins.toJSON cfg.allowedHosts}
+      # Default database engine is SQLite. So one can just check out code,
+      # install requirements.txt and do manage.py runserver and it works
+      DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
+      DATABASES = {
+          "default": {
+          "ENGINE": "django.db.backends.sqlite3",
+          "NAME": "${cfg.dataDir}/healthchecks.sqlite",
+        }
+      }
+      with open("${cfg.secretKeyFile}", "r") as file:
+          SECRET_KEY = file.readline()
+      DEBUG = ${boolToPython cfg.debug}
+      REGISTRATION_OPEN = ${boolToPython cfg.registrationOpen}
+
+      ${cfg.extraConfig}
+    '';
+  };
+  pkg = (pkgs.healthchecks.overrideAttrs (old: {
+    installPhase = old.installPhase + ''
+      ln -s ${localSettingsFile} $out/opt/healthchecks/hc/local_settings.py
+    '';
+  }));
+  healthchecksManageScript = with pkgs; (writeScriptBin "healthchecks" ''
+    #!${stdenv.shell}
+    export PYTHONPATH=${pkg.pythonPath}
+    sudo -u healthchecks ${pkg}/bin/healthchecks "$@"
```

```suggestion
    ${pkg}/bin/healthchecks "$@"
```
We should already be that user when executing

---

In systemd services: yes, but it's also added to `environment.systemPackages` to run maintenance tasks manually and i think it's useful to do this with the right user by default, no?
(that's also copied from netbox, so we might want to adapt it there as well if so)